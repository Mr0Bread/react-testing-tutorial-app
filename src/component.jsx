import React, { useEffect, useState } from 'react';

export const useMockRequest = (delay) => {
	const [isLoading, setIsLoading] = useState(true);

	useEffect(() => setTimeout(() => setIsLoading(false), delay), []);

	return [isLoading];
}

export default function TestComponent({ children }) {
	const [isLoading] = useMockRequest(100);

	if (isLoading) {
		return (
			<div>Loading</div>
		);
	}

	return (
		<div>
			{ children }
		</div>
	);
};
