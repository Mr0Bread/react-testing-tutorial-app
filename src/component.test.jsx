import React from 'react';
import TestComponent from './component';
import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import * as TestComponentModule from './component';
import '@testing-library/jest-dom';

describe('TestComponent', () => {
    it('should match snapshot', () => {
        expect(
            create(<TestComponent />)
                .toJSON()
        )
            .toMatchSnapshot();
    });

    it('should render properly', async () => {
        render(
            <TestComponent>
                <div>Node</div>
            </TestComponent>
        );

        expect(screen.getByText('Loading'))
            .toBeInTheDocument();

        expect(await screen.findByText('Node'))
            .toBeInTheDocument();
    });
});

describe('useMockRequest', () => {
    it('should change state after given delay', () => {
        const useMockRequestSpy = jest.spyOn(TestComponentModule, 'useMockRequest');

        render(
            <TestComponent>
                <div>Node</div>
            </TestComponent>
        );

        expect(useMockRequestSpy).toBeCalled();
    })
});
