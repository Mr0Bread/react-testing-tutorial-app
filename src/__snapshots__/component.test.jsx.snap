// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent should match snapshot 1`] = `
<div>
  Loading
</div>
`;
